golang/checksum.go

```diff
@@ -59,7 +63,7 @@ func (t ChecksumType) New() Checksum {
 	case ChecksumTypeNone:
 		return nullChecksum{}
 	case ChecksumTypeCrc32:
```

Comment:
sadly this needs to maintain the old adler-32 setting, there's a new constat (0x03) for castagnoli